R/shiny.R

```diff
@@ -1041,7 +1041,28 @@ ShinySession <- R6Class(
       return(private$inputReceivedCallbacks$register(callback))
     },
     unhandledError = function(e) {
-      self$close()
+      # always close the websocket
+      on.exit({
+        self$close()
+      })
+
+      handler <- getOption("shiny.unhandled.error", NULL)
+
+      # no handler: return early
+      if(!is.function(handler)) {
+        return()
+      }
+
+      # we check whether the handler can accept the error
+      # as parameter.
+      formals <- formals(handler)
+
+      if(length(formals) == 0){
+        handler()
+        return()
+      }
+
+      handler(e)
```

Comment:
Should probably `tryCatch` here so that the handler can't cause an exception that goes up out of `unhandledError`.

Reply:
How would developers catch errors in the handler? Should an error lead to a message?